zebra-rpc/src/methods/get_block_template.rs

```diff
@@ -139,8 +150,106 @@ where
         }
         .boxed()
     }
+
+    fn get_block_template(&self) -> BoxFuture<Result<GetBlockTemplate>> {
+        async move {
+            let empty_string = String::from("");
+
+            // Returns empty `GetBlockTemplate`
+            Ok(GetBlockTemplate {
+                capabilities: vec![],
+                version: 0,
+                previousblockhash: empty_string.clone(),
+                blockcommitmentshash: empty_string.clone(),
+                lightclientroothash: empty_string.clone(),
+                finalsaplingroothash: empty_string.clone(),
+                defaultroots: DefaultRoots {
+                    merkleroot: empty_string.clone(),
+                    chainhistoryroot: empty_string.clone(),
+                    authdataroot: empty_string.clone(),
+                    blockcommitmentshash: empty_string.clone(),
+                },
+                transactions: vec![],
+                coinbasetxn: Coinbase {},
+                longpollid: empty_string.clone(),
+                target: empty_string.clone(),
+                mintime: 0,
+                mutable: vec![],
+                noncerange: empty_string.clone(),
+                sigoplimit: 0,
+                sizelimit: 0,
+                curtime: 0,
+                bits: empty_string,
+                height: 0,
+            })
+        }
+        .boxed()
+    }
+}
+/// Documentation to be added after we document all the individual fields.
+#[derive(Clone, Debug, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct GetBlockTemplate {
+    /// Add documentation.
+    pub capabilities: Vec<String>,
+    /// Add documentation.
+    pub version: usize,
+    /// Add documentation.
+    pub previousblockhash: String,
+    /// Add documentation.
+    pub blockcommitmentshash: String,
+    /// Add documentation.
+    pub lightclientroothash: String,
+    /// Add documentation.
+    pub finalsaplingroothash: String,
+    /// Add documentation.
+    pub defaultroots: DefaultRoots,
+    /// Add documentation.
+    pub transactions: Vec<Transaction>,
+    /// Add documentation.
+    pub coinbasetxn: Coinbase,
+    /// Add documentation.
+    pub longpollid: String,
+    /// Add documentation.
+    pub target: String,
+    /// Add documentation.
+    pub mintime: u32,
+    /// Add documentation.
+    pub mutable: Vec<String>,
+    /// Add documentation.
+    pub noncerange: String,
+    /// Add documentation.
+    pub sigoplimit: u32,
+    /// Add documentation.
+    pub sizelimit: u32,
+    /// Add documentation.
+    pub curtime: u32,
```

Comment:
Optional: For some other RPCs, we've used the Rust standard where words are separated by underscores, then renamed the RPC when serializing.

Did you want to do that here?

https://github.com/ZcashFoundation/zebra/blob/a2a75059d39dcd9aaabb9609a84e0dc59b966535/zebra-rpc/src/methods.rs#L1009

Reply:
Yea, we definitely want to do this. https://github.com/ZcashFoundation/zebra/pull/5462/commits/5756a77f89d8597ce146da071878d978c4706711
Thanks.